File: legacy/2026-05-cleanup/wt-editor-routes/engine/pages/approvals_page.py
"""ApprovalsPage — TestwrightAI Onaylar ekranı Page Object."""
from playwright.sync_api import Page, expect
from pages.base_page import BasePage


class ApprovalsPage(BasePage):
    """TestwrightAI Onaylar sayfası."""

    HEADING = '[data-testid="approvals-heading"]'
    APPROVE_BTN = '[data-testid="approvals-btn-approve"]'
    REJECT_BTN = '[data-testid="approvals-btn-reject"]'

    def __init__(self, page: Page, project_id: str, base_url: str = ""):
        super().__init__(page)
        self._project_id = project_id
        self._base_url = base_url

    def goto(self):
        self.navigate(f"{self._base_url}/p/{self._project_id}/approvals")
        return self

    def approval_item(self, approval_id: str):
        return self.page.locator(f'[data-testid="approvals-item-{approval_id}"]')

    def click_approval(self, approval_id: str):
        self.approval_item(approval_id).click()
        return self

    def approve(self):
        self.page.locator(self.APPROVE_BTN).click()
        return self

    def reject(self):
        self.page.locator(self.REJECT_BTN).click()
        return self

    def assert_page_loaded(self):
        expect(self.page.locator(self.HEADING)).to_be_visible()
        return self

File: e2e/pages/approvals.page.ts
import { expect } from "@playwright/test";
import { BasePage } from "./base.page";

export class ApprovalsPage extends BasePage {
  readonly url: RegExp = /\/p\/[^/]+\/approvals$/;

  // ── Locators ───────────────────────────────────────────────────────────────
  get heading() {
    return this.text("Onaylar");
  }
  get approveButton() {
    return this.testId("approvals-btn-approve");
  }
  get rejectButton() {
    return this.testId("approvals-btn-reject");
  }

  approvalItem(title: string) {
    return this.text(title);
  }

  // ── Actions ────────────────────────────────────────────────────────────────
  async approve() {
    if (!(await this.approveButton.isVisible({ timeout: 1_000 }).catch(() => false))) {
      await this.page.locator("[data-testid^='approvals-card-']").first().click();
      await expect(this.approveButton).toBeVisible({ timeout: 10_000 });
    }
    const decidePromise = this.page.waitForResponse(
      (r) =>
        r.url().includes("/approvals/") &&
        r.url().includes("/decide") &&
        r.request().method() === "POST"
    );
    await this.approveButton.click();
    const res = await decidePromise;
    expect(res.ok()).toBeTruthy();
  }

  async reject() {
    if (!(await this.rejectButton.isVisible({ timeout: 1_000 }).catch(() => false))) {
      await this.page.locator("[data-testid^='approvals-card-']").first().click();
      await expect(this.rejectButton).toBeVisible({ timeout: 10_000 });
    }
    const decidePromise = this.page.waitForResponse(
      (r) =>
        r.url().includes("/approvals/") &&
        r.url().includes("/decide") &&
        r.request().method() === "POST"
    );
    await this.rejectButton.click();
    const res = await decidePromise;
    expect(res.ok()).toBeTruthy();
  }

  // ── Assertions ─────────────────────────────────────────────────────────────
  async assertApprovalVisible(title: string) {
    await expect(this.approvalItem(title)).toBeVisible({ timeout: 10_000 });
  }

  async assertStatusApproved() {
    await expect(
      this.page.getByTestId("kanban-column-approved").locator("[data-testid^='approvals-card-']")
    ).toHaveCount(1, { timeout: 10_000 });
  }

  async assertStatusRejected() {
    await expect(
      this.page.getByTestId("kanban-column-rejected").locator("[data-testid^='approvals-card-']")
    ).toHaveCount(1, { timeout: 10_000 });
  }
}
